vars/utils.groovy

```diff
@@ -9,6 +9,77 @@ import java.text.SimpleDateFormat
 
 import org.kohsuke.github.GitHub
 
+
+// Pytest exit codes
+// https://docs.pytest.org/en/stable/reference.html#pytest.ExitCode
+int PYTEST_EXIT_OK              = 0
+int PYTEST_EXIT_TESTS_FAILED    = 1
+int PYTEST_EXIT_INTRRUPTED      = 2
+int PYTEST_EXIT_INTERNAL_ERROR  = 3
+int PYTEST_EXIT_USAGE_ERROR     = 4
+int PYTEST_EXIT_NO_TESTS        = 5
+
+// Minimum version of pytest capable of emitting reliable exit codes
+int PYTEST_EXIT_CAPABLE = [5, 0]
+
+
+// Determine if a program is available on $PATH
+//
+// @param name      String      program name
+// @return          int         Non-zero on failure, zero on success
+int programExists(String name) {
+    // Sanitize input
+    name = name.split('\\ |\\;|\\||\\&\\&?|\\/|\\\')[0]
+    // Find program, return status
+    return sh(script: "which ${name}", label: "Check program exists: ${name}", returnStatus: true)
+}
+
+
+// Test minimum pytest version available
+//
+// @param   needs   List        e.g. [1, 2, 3]
+// @return          Boolean     `needs` >= the system pytest version
+def pytestVersionMin(def needs) {
```

Comment:
This seems a bit complicated for what it does.
Wouldn't something like 
```
too_old = ['0','1','2','3','4']
major = ver.split(' ')[1].trim()[0]
if (too_old.contains(major)) {
  retval = false
} else {
  retval = true
}
```
be shorter and easier to read?

Reply:
I suppose that's easier on the eyes. :grin: 

I was going for something a bit more generic with my "I wish this was `C`" implementation here...

```groovy
ver = "pytest 1.2.3\n"

tests = [
    pytestVersionMin([0]),       // false
    pytestVersionMin([1,1]),     // false
    pytestVersionMin([1,1,2]),   // false
    pytestVersionMin([1,2,2]),   // false
    
    pytestVersionMin([1,2,3,4]), // true (parsed up to 1.2.3)
    pytestVersionMin([1,2,3]),   // true
    pytestVersionMin([1,2]),     // true
    pytestVersionMin([1]),       // true
]
```

Reply:
I agree that it allows for greater granularity, but it doesn't seem that the granularity is needed in this case.
I understand the situation to be:
pytest >= 5.0 supports the desired status codes
pytest < 5.0 does not
Is that actually the case?
The set of pytest versions that don't support status codes is static. A membership check allows those versions to be ignored. Every other version behaves as needed for status reporting purposes.